src/worker.js

```diff
@@ -191,8 +191,14 @@ this.onmessage = function(e) {
       assert(selfThreadId);
       assert(parentThreadId);
       assert(STACK_BASE != 0);
-#if !WASM_BACKEND
-      assert(STACK_MAX > STACK_BASE);
+#if WASM_BACKEND
+      assert(max === e.data.stackBase);
+      assert(top > max);
```

Comment:
I think it does still make sense to rename 'top' and 'max' as @vargaz suggested.

Reply:
I opened #8843 now so we don't forget about that.